packages/flutter/lib/src/material/dialog.dart

```diff
@@ -481,12 +486,15 @@ class _DialogRoute<T> extends PopupRoute<T> {
 Future<T> showDialog<T>({
   @required BuildContext context,
   bool barrierDismissible: true,
-  @required Widget child,
+  @deprecated Widget child,
```

Comment:
`@Deprecated('Instead of using the "child" argument, return the child from a closure provided to the "builder" argument. This will ensure that the BuildContext is appropriate for widgets built in the dialog.')`

Reply:
Done